manifests/client/monitor.pp

```diff
@@ -12,34 +12,72 @@
 # @param files A hash of filenames as key and sources as values to add to the xymon files
 # @param sudo A sudo::conf hash with sudo definitions the xymon user should be allowed to use
 # @param packages A puppet package hash with packages that are required for the monitor to work
+# @param logrotate A hash containing definitions to configure logfile rotation
 define xymon::client::monitor (
   String $script_source,
   String $clientlaunch_config,
   String $files_path,
   String $xymon_user,
   String $xymon_group,
   String $xymon_service,
-  String $interval                 = '5m',
-  Array[String] $arguments         = [],
-  Optional[String] $require_fqdn   = undef,
-  Optional[Hash] $files = undef,
-  Optional[Hash] $sudo             = undef,
-  Optional[Hash] $packages         = undef,
+  String $interval                        = '5m',
+  Array[String] $arguments                = [],
+  Optional[String] $require_fqdn          = undef,
+  Optional[Hash] $files                   = undef,
+  Optional[Hash] $sudo                    = undef,
+  Optional[Hash] $packages                = undef,
+  Optional[Hash] $logrotate               = undef,
 ) {
   if (!$require_fqdn or $facts['fqdn'] == $require_fqdn) {
     if ($files) {
-      $files.each |$value| {
-        ensure_resource(
-          'file',
-          "${files_path}/${value[0]}",
-          {
-            owner  => $xymon_user,
-            group  => $xymon_group,
-            source => $value[1],
-            before => Service[$xymon_service]
-          }
-        )
+      $files.each |String $key, Hash $value| {
+        $mode = !empty($value[mode]) ? {
```

Comment:
It's common practice to use positive tests, because negative checks tend to look more complex. So please test against empty($value[mode}) (and change the values of the selector accordingly)

Reply:
I will, thanks. I just copied this from... uhm, someone :)

Reply:
Fixed now and spotted also another mistake in the files block.